db/background_mgr_attachment_migration.go

```diff
@@ -0,0 +1,288 @@
+
+	select {
+	case <-doneChan:
+		base.InfofCtx(ctx, base.KeyAll, "[%s] Finished migrating attachment metadata from sync data to global sync data. %d/%d docs changed", migrationLoggingID, a.DocsChanged.Value(), a.DocsProcessed.Value())
+		err = dcpClient.Close()
+		if err != nil {
+			base.WarnfCtx(ctx, "[%s] Failed to close attachment migration DCP client! %v", migrationLoggingID, err)
+			return err
+		}
+		if processFailure != nil {
+			return processFailure
+		}
+		// set sync info here
+		for _, collectionID := range collectionIDs {
+			dbc, ok := db.CollectionByID[collectionID]
+			if !ok {
+				base.WarnfCtx(ctx, "[%s] Completed attachment migration, but unable to update syncInfo for collection %v (not found)", migrationLoggingID, collectionID)
+				continue
+			}
+			if err := base.SetSyncInfoMetaVersion(dbc.dataStore, base.ProductMajorVersionInteger); err != nil {
+				base.WarnfCtx(ctx, "[%s] Completed attachment migration, but unable to update syncInfo for collection %v: %v", migrationLoggingID, collectionID, err)
+			}
+		}
+	case <-terminator.Done():
+		base.DebugfCtx(ctx, base.KeyAll, "[%s] Terminator closed. Ending Attachment Migration process.", migrationLoggingID)
+		err = dcpClient.Close()
+		if err != nil {
+			base.WarnfCtx(ctx, "[%s] Failed to close attachment migration DCP client! %v", migrationLoggingID, err)
+			return err
+		}
+		if processFailure != nil {
+			return processFailure
+		}
+		err = <-doneChan
+		if err != nil {
+			return err
+		}
+		base.InfofCtx(ctx, base.KeyAll, "[%s] Attachment Migration was terminated. Docs changed: %d Docs Processed: %d", migrationLoggingID, a.DocsChanged.Value(), a.DocsProcessed.Value())
+	}
+	return nil
+}
+
+func (a *AttachmentMigrationManager) SetStatus(docChanged, docProcessed int64) {
+
+	a.DocsChanged.Set(docChanged)
+	a.DocsProcessed.Set(docProcessed)
+}
+
+func (a *AttachmentMigrationManager) ResetStatus() {
+	a.DocsProcessed.Set(0)
+	a.DocsChanged.Set(0)
+}
+
+func (a *AttachmentMigrationManager) GetProcessStatus(status BackgroundManagerStatus) ([]byte, []byte, error) {
+	a.lock.RLock()
+	defer a.lock.RUnlock()
+
+	response := AttachmentMigrationManagerResponse{
+		BackgroundManagerStatus: status,
+		MigrationID:             a.MigrationID,
+		DocsChanged:             a.DocsChanged.Value(),
+		DocsProcessed:           a.DocsProcessed.Value(),
+	}
+	statusJSON, err := base.JSONMarshal(response)
+	if err != nil {
+		return nil, nil, err
+	}
+	return statusJSON, nil, err
+}
+
+func getMigrationDCPClientOptions(collectionIDs []uint32, groupID, prefix string) *base.DCPClientOptions {
+	checkpointPrefix := prefix + "att_migration:"
+	clientOptions := &base.DCPClientOptions{
+		OneShot:           true,
+		FailOnRollback:    false,
+		MetadataStoreType: base.DCPMetadataStoreCS,
+		GroupID:           groupID,
+		CollectionIDs:     collectionIDs,
+		CheckpointPrefix:  checkpointPrefix,
+	}
+	return clientOptions
+}
+
+func getCollectionIDs(db *Database) []uint32 {
```

Comment:
This looks like something that's worth defining as `func (db *Database) GetCollectionIDs() []uint32` in database.go for future reuse.